test/runtests.jl

```diff
@@ -17,7 +22,7 @@ using Base.Test, Missings, Compat
     @test promote_type(Union{Int, Missing}, Union{Int, Missing}) == Union{Int, Missing}
     @test promote_type(Union{Float64, Missing}, Union{String, Missing}) == Any
     @test promote_type(Union{Float64, Missing}, Union{Int, Missing}) == Union{Float64, Missing}
-    @test promote_type(Union{Void, Missing, Int}, Float64) == Any
+    # @test_broken promote_type(Union{Nothing, Missing, Int}, Float64) == Any
```

Comment:
@nalimilan, are we maybe missing a `promote_rule` in the Base definitions? I'm seeing
```julia
julia> promote_type(Union{Nothing, Missing, Int}, Float64)
Union{Missing, Float64}
```
on 0.7 (as of today).

Reply:
That's https://github.com/JuliaLang/julia/pull/24653#discussion_r151874283. See also https://github.com/JuliaLang/julia/issues/25240.

Reply:
But why comment this out? Isn't it still broken?

Reply:
It's not broken on 0.6 though

Reply:
How did tests pass then? :-/

Reply:
OK. Why not make it conditional on the version then?